### DISCUSSION TITLE
Are we supposed to install Blade Icons UI Kit?

### DISCUSSION THREAD

**Discussion by myregistration** [filamentphp/filament#12378] [Help] [answered]
### Package

Other

### Package Version

v3

### How can we help you?

It appears Filament already has access to some icons in the UI.  The instructions say to publish the Blade Icon config file and uncomment the default array within sets , but there is no resources/svg directory and I don't see one in the vendor directory either.  The documentation makes it sound like the icon package is installed automatically when you install Filament.  Are we supposed to install Blade Icons ourselves?  I can do it, as I've done it before, I just don't want duplicate files as there are a lot of icons in general.

**Comment by myregistration** [comment, marked as answer]
You can follow instructions at https://github.com/blade-ui-kit/blade-icons

In order to use the blade icons I installed the Blade UI Kit for Icons 
``composer require blade-ui-kit/blade-icons``

Important: Make sure that the path defined for this icon set exists. By default it's resources/svg. Your SVG icons will need to be placed in this directory.

published the config file and uncommented the default array within the set config
``php artisan vendor:publish --tag=blade-icons``

Install icon package, for example heroicons.  You can also publish a relative config file.
https://github.com/blade-ui-kit/blade-heroicons

You can see a list of available packages at https://github.com/blade-ui-kit/blade-icons?tab=readme-ov-file#icon-packages

If you have missing icons, like social media icons, you can copy them into the default directory set in the blade config, for example /resources/svg

After you install a package you can access the icon svg using something like ``<x-heroicon-c-camera/>``

I'm still not sure if Filament installs the Blade Icon UI Kit automatically since they refer to heroicons, but if I determine it I will update this comment.

**Comment by micraux** [comment]
As far as I can tell, in [https://filamentphp.com/docs/3.x/support/icons#overview](https://filamentphp.com/docs/3.x/support/icons#overview):
"_To render icons, we use the [Blade Icons](https://github.com/blade-ui-kit/blade-icons) package from Blade UI Kit._"

So I guess you don't need to `composer require blade-ui-kit/blade-icons`

As written in the Filament documentation, you can just start with `php artisan vendor:publish --tag=blade-icons`

**Comment by wickham-c** [comment]
If you look at the Filament [composer.lock](https://github.com/filamentphp/filament/blob/3.x/composer.lock) file, and search for `blade-ui-kit/blade-heroicons`, you will see that it is a package which `filament/support` depends upon and thus is installed already. Therefore, you can use heroicons without installing any packages. Incidentally, `blade-ui-kit/blade-heroicons` depends upon the `blade-ui-kit/blade-icons` package, so that package is also installed without you specifically needing to do anything.

I believe that you only need to publish the Blade Icons configuration file if you wish to use _your own_ custom icons in Filament (and hence these instructions are included in a section headed [Using custom SVGs as icons](https://filamentphp.com/docs/3.x/support/icons#using-custom-svgs-as-icons)).

I have **not** run the `vendor:publish` command and do **not** have a "config/blade-icons.php" file **nor** a "resources/svg" directory.

My process for including icons from other libraries is...

1. Search for an appropriate icon on this site: https://blade-ui-kit.com/blade-icons
2. Click the chosen icon to see details specific to that icon (e.g. https://blade-ui-kit.com/blade-icons/iconic-user-check ).
3. Copy the "**Install through composer**" code snippet on the above icon page (which will begin with `composer require`) and paste that command in your terminal (e.g. `composer require itsmalikjones/blade-iconic`). This will add the icon library package to your "composer.json" and "composer.lock" files.
4. Copy the string argument from the "**Use it with a helper**" code snippet from the above page and in your Filament project set the appropriate icon property to this string value (e.g. `protected static ?string $navigationIcon = 'iconic-user-check';`).

**Reply by denizaygundev** [reply]
Thanks @wickham-c this worked for me also.

I just wonder if this is optimal and the recommended way, I did some googling and couldn't find any other alternative approaches besides publishing the `blade-icons.php` config and actually downloading the SVGs manually but this doesn't seem desirable. 

@danharrin sorry to tag you but just wondering if you could shed any insights on this?

**Reply by danharrin** [reply]
What are you trying to do?

**Reply by denizaygundev** [reply]
@danharrin just to use some of the other blade-icon packs
`protected static ?string $navigationIcon = 'healthicons-o-prescription-document';`

This only works after running `composer require troccoli/blade-health-icons` but maybe this is intentional? 

The docs don't cover this from what I can see: https://filamentphp.com/docs/3.x/support/icons

**Reply by danharrin** [reply]
Yes, you need to install icon packs if you want to use them, we only install Heroicons by default. No need to publish config or download SVGs manually.

**Reply by denizaygundev** [reply]
Brilliant, thanks for confirming @danharrin 

I just wanted to be sure that their wasn't a more "proper" and optimal way 

(On a side note, absolutely love Filament and everything you and your team are doing - very excited for v4!)
